bookshelf/src/main/webapp/base.jsp

```diff
@@ -34,9 +34,11 @@ Copyright 2015 Google Inc. All Rights Reserved.
         </div>
         <ul class="nav navbar-nav">
           <li><a href="/">Books</a></li>
-          <li><a href="/books/mine">My Books</a></li>
+          <div class="form-group ${isAuthConfigured ? '' : 'hidden'}">
```

Comment:
actually the webservlet annotation has a loadonstartup = 1. this guarantees that the servlet is loaded first.